spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java

```diff
@@ -220,6 +220,17 @@ private void registerBuiltInBeans(BeanDefinitionRegistry registry) {
 						"there is no jayway json-path.jar on the classpath.");
 			}
 
+			try {
+				ClassUtils.forName("com.jayway.jsonpath.Predicate", this.classLoader);
+			}
+			catch (ClassNotFoundException e) {
+				jsonPathClass = null;
+				logger.error("SpEL function '#jsonPath' isn't registered. " +
```

Comment:
I think we should only log this as an error if `JsonPath` __is__ on the CP - they might not be using json path at all.